File: ballast-core/src/typescript/services/game-service.ts
import { IGame } from '../models/game';
import { IVesselMoveRequest } from '../value-objects/vessel-move-request';
import { IDisposable } from '../interfaces/disposable';
import { ICreateGameOptions } from '../value-objects/create-game-options';
import { IAddPlayerOptions } from '../value-objects/add-player-options';
import { IRemovePlayerOptions } from '../value-objects/remove-player-options';
import { IVessel } from '../models/vessel';

export interface IGameService extends IDisposable {
    getAllGamesAsync(): Promise<IGame[]>;
    getGameAsync(gameId: string): Promise<IGame>;
    getTestGameIdAsync(): Promise<string>;
    createGameAsync(options: ICreateGameOptions): Promise<IGame>;
    startGameAsync(gameId: string): Promise<IGame>;
    endGameAsync(gameId: string): Promise<IGame>;
    deleteGameAsync(gameId: string): Promise<void>;
    addPlayerToGameAsync(options: IAddPlayerOptions): Promise<IGame>;
    removePlayerFromGameAsync(options: IRemovePlayerOptions): Promise<IGame>;
    addPlayerToVesselAsync(options: IAddPlayerOptions): Promise<IVessel>;
    removePlayerFromVesselAsync(options: IRemovePlayerOptions): Promise<IVessel>;
    addPlayerToVesselRoleAsync(options: IAddPlayerOptions): Promise<IVessel>;
    removePlayerFromVesselRoleAsync(options: IRemovePlayerOptions): Promise<IVessel>;
    moveVesselAsync(request: IVesselMoveRequest): Promise<void>;
}

File: ballast-core/dotnet/src/application/services/GameService.cs
using Ballast.Core.Application.Models;
using System;
using System.Collections.Generic;
using System.Threading.Tasks;

namespace Ballast.Core.Application.Services
{
    public interface IGameService : IDisposable
    {
        Task<string> GetTestGameIdAsync();
        Task<IEnumerable<GameDto>> GetAllGamesAsync();
        Task<GameDto> GetGameAsync(string gameId);
        Task<GameDto> CreateGameAsync(CreateGameOptions options);
        Task<GameDto> StartGameAsync(string gameId);
        Task<GameDto> EndGameAsync(string gameId);
        Task DeleteGameAsync(string gameId);
        Task<GameDto> AddPlayerToGameAsync(AddPlayerOptions options);
        Task<GameDto> RemovePlayerFromGameAsync(RemovePlayerOptions options);
        Task<VesselDto> AddPlayerToVesselAsync(AddPlayerOptions options);
        Task<VesselDto> RemovePlayerFromVesselAsync(RemovePlayerOptions options);
        Task<VesselDto> AddPlayerToVesselRoleAsync(AddPlayerOptions options);
        Task<VesselDto> RemovePlayerFromVesselRoleAsync(RemovePlayerOptions options);
        Task<VesselDto> MoveVesselAsync(VesselMoveRequest request);
    }
}